src/mixins/form-text.js

```diff
@@ -105,68 +106,43 @@ export default {
+      if (this.number) {
+        const num = parseFloat(value)
+        value = isNaN(num) ? value : num
+      }
+      if (applyFormatter && isFunction(this.formatter)) {
         value = this.formatter(value, evt)
       }
       return value
     },
-    updateValue(value, lazyUpdate = false) {
-      value = this.stringifyValue(value)
-      const $input = this.$refs.input
-      if (value !== this.localValue) {
-        if (lazyUpdate) {
-          // If lazy update, we just update the input's value
-          // and will emit v-model update on change/blur later
-          if ($input && $input.value !== value) {
-            // Apply any formating to input's value,
-            // but only if the value has changed
-            /* istanbul ignore next: hard to test */
-            $input.value = value
-          }
-        } else {
-          // Update the v-model (if not a lazy update)
-          // Keep the input set to the value before any modifiers
-          this.localValue = value
-          if (this.number) {
-            // Emulate `.number` modifier behaviour
-            const num = parseFloat(value)
-            value = isNaN(num) ? value : num
-          } else if (this.trim) {
-            // Emulate `.trim` modifier behaviour
-            value = value.trim()
-          }
-          this.$emit('update', value)
-        }
-      } else if ($input && $input.value !== value) {
-        // When the `localValue` hasn't changed but the actual input value
-        // is out of sync, make sure to change it to the given one
-        // Usually caused by browser autocomplete and how it triggers the
-        // change or input event, or depending on the formatter function
-        // https://github.com/bootstrap-vue/bootstrap-vue/issues/2657
-        // https://github.com/bootstrap-vue/bootstrap-vue/issues/3498
-        /* istanbul ignore next: hard to test */
-        this.$refs.input.value = value
-      }
-    },
     onInput(evt) {
       // `evt.target.composing` is set by Vue
       // https://github.com/vuejs/vue/blob/dev/src/platforms/web/runtime/directives/model.js
       /* istanbul ignore if: hard to test composition events */
       if (evt.target.composing) {
         return
       }
-      const formatted = this.getFormatted(evt.target.value, evt)
+      const value = evt.target.value
+      const formattedValue = this.formatValue(value, evt, !this.lazyFormatter)
       // Exit when the `formatter` function strictly returned `false`
       // or prevented the input event
-      if (formatted === false || evt.defaultPrevented) {
-        /* istanbul ignore next */
+      /* istanbul ignore next */
+      if (formattedValue === false || evt.defaultPrevented) {
         evt.preventDefault()
         return
       }
-      this.updateValue(formatted, this.lazy)
-      this.$emit('input', formatted)
+      this.localValue = value
+      if (!this.lazy) {
+        this.$emit('update', formattedValue)
+      }
```

Comment:
And typically the formatted value (when formatted) should update the actual value of the input (this.$input.value = formatted).  This will work if the user has a `v-model` applied, but it should also work when `v-model` isn't explicitly used, so the input's DOM prop value should be updated.